File: n26/n26.go
package n26

import (
	"rahul-rakshit/formatbankcsv/csv"
	"rahul-rakshit/formatbankcsv/utils"
)

type Formatter struct{}

func (f Formatter) ReadCsv(inputPath string) ([][]string, error) {
	delimiter := ","
	skipLines := 0

	return csv.ReadCsv(inputPath, delimiter, skipLines)
}

func (f Formatter) Format(csvData [][]string, fromDate string, toDate string) ([][]string, error) {
	formatted, formatErr := FormatN26(csvData)
	if formatErr != nil {
		return [][]string{}, formatErr
	}

	filtered, filterErr := utils.FilterByDate(formatted, fromDate, toDate)
	if filterErr != nil {
		return [][]string{}, filterErr
	}

	return filtered, nil
}

func (f Formatter) WriteCsv(data [][]string, outputPath string) error {
	delimiter := ","

	return csv.WriteCsv(data, outputPath, delimiter)
}

File: n26/n26.ts
import { readCsv, writeCsv } from "../csv/csv.ts";
import { filterByDate } from "../utils/filterByDate.ts";
import { formatN26 } from "./formatN26.ts";

export class N26Formatter {
  async readCsv(inputPath: string): Promise<string[][]> {
    const delimiter = ",";
    const skipLines = 0;
    return await readCsv(inputPath, delimiter, skipLines);
  }

  format(csvData: string[][], fromDate?: string, toDate?: string): string[][] {
    const formatted = formatN26(csvData);
    if (fromDate || toDate) {
      return filterByDate(formatted, fromDate || "", toDate || "");
    }
    return formatted;
  }

  async writeCsv(data: string[][], outputPath: string): Promise<void> {
    const delimiter = ",";
    return await writeCsv(data, outputPath, delimiter);
  }
}
